sklearn/manifold/t_sne.py

```diff
@@ -767,10 +770,10 @@ def _fit(self, X, skip_num_points=0):
             pca = PCA(n_components=self.n_components, svd_solver='randomized',
                       random_state=random_state)
             X_embedded = pca.fit_transform(X)
-        elif isinstance(self.init, np.ndarray):
-            X_embedded = self.init
         elif self.init == 'random':
             X_embedded = None
+        elif isinstance(self.init, np.ndarray):
```

Comment:
This change is irrelevant because the numpy array check has to come first to avoid throwing a FutureWarning per the comments below.